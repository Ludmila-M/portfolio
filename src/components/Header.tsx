import React, { useState } from "react";
import Switch from "react-switch";
import "../styles/Header.scss";

interface HeaderProps {
  switchTheme: () => void;
}

const Header: React.FC<HeaderProps> = ({ switchTheme }) => {
  const [isChecked, setChecked] = useState(false);
  const handleClick = () => {
    switchTheme();
    setChecked(!isChecked);
  };
  return (
    <div className="app__header col-12">
      <Switch onChange={handleClick} checked={isChecked} className="app__header-switch" />
    </div>
  );
};

export default Header;
