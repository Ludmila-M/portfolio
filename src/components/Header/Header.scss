.header {
    display: flex;
    justify-content: flex-end;
    align-items: center;
    justify-content: space-between;
    transition: background-color 0.5s ease;
}

.header__button {
    cursor: pointer;
    transition: all .2s ease-in-out;
    border: none;
    background-color: transparent;
    font-size: 2rem;
    &:hover {
        transform: scale(1.2);
    }
}

.header__nav {
    display: flex;
    gap: 2rem;
    list-style-type: none;
    font-size: 2rem;
    margin: 0;
}

.header__nav-category {
    cursor: pointer;
}

.header__switch {
    align-self: center;
}