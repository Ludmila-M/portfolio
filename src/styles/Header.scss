.app__header {
    display: flex;
    justify-content: flex-end;
    height: 5vh;
    transition: background-color 0.5s ease;
}

.app__header-switch {
    align-self: center;
}