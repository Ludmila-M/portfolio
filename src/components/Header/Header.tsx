import React, { useState } from "react";
import Switch from "react-switch";
import "./Header.scss";
import { Categories } from "../../enums/categories";
import { basicData } from "../../data";

const navCategories = [
  Categories.BIO,
  Categories.EXPIERENCE,
  Categories.PROJECTS,
  Categories.CONTACT,
];

interface HeaderProps {
  switchTheme: () => void;
  setPickedCategory: (category: string) => void;
  pickedCategory: string;
}

const Header: React.FC<HeaderProps> = ({
  switchTheme,
  setPickedCategory,
  pickedCategory,
}) => {
  const [isChecked, setChecked] = useState(false);
  const handleClick = () => {
    switchTheme();
    setChecked(!isChecked);
  };
  return (
    <div className="app__header header col-12">
      <span
        onClick={() => setPickedCategory(Categories.INTRO)}
        className="header__button"
      >
        {basicData.initials}
      </span>
      <ul className="header__nav">
        {navCategories.map((category: string, index: number) => (
          <li
            key={index}
            className={`${
              pickedCategory === category
                ? "header__nav-category--active"
                : "header__nav-category"
            }`}
            onClick={() => setPickedCategory(category)}
          >
            {category}
          </li>
        ))}
      </ul>
      <Switch
        className="header__switch"
        onChange={handleClick}
        checked={isChecked}
        onColor="#e53170"
        offColor="#ff8906"
      />
    </div>
  );
};

export default Header;
