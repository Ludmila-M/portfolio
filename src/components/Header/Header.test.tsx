import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import Header from "./Header";

const HeaderProps = {
  switchTheme: jest.fn(),
  setPickedCategory: jest.fn(),
  pickedCategory: "CONTACT",
};

describe("my beverage", () => {
  test("matches snapshot", () => {
    const header = render(<Header {...HeaderProps} />);
    expect(header).toMatchSnapshot();
  });
  test("switches theme", () => {
    const { container } = render(<Header {...HeaderProps} />);
    const switcher = container.querySelector(`.react-switch-bg`)!;
    fireEvent.click(switcher);
    expect(HeaderProps.switchTheme).toBeCalledTimes(1);
  });
  test("changes category", () => {
    const { container } = render(<Header {...HeaderProps} />);
    const category = container.querySelectorAll(`.header__nav-category`)!;
    fireEvent.click(category[0]);
    expect(HeaderProps.setPickedCategory).toBeCalledTimes(1);
  });
});
