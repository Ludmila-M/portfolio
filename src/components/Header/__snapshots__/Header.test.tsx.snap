// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my beverage matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app__header header col-12"
      >
        <span
          class="header__button"
        >
          LM
        </span>
        <ul
          class="header__nav"
        >
          <li
            class="header__nav-category"
          >
            Bio
          </li>
          <li
            class="header__nav-category"
          >
            Experience
          </li>
          <li
            class="header__nav-category"
          >
            Projects
          </li>
          <li
            class="header__nav-category"
          >
            Contact
          </li>
        </ul>
        <div
          class="header__switch"
          style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 14px; transition: opacity 0.25s; user-select: none;"
        >
          <div
            class="react-switch-bg"
            style="height: 28px; width: 56px; margin: 0px; position: relative; background: rgb(255, 137, 6); border-radius: 14px; cursor: pointer; transition: background 0.25s;"
          >
            <div
              style="height: 28px; width: 30px; position: relative; opacity: 0; pointer-events: none; transition: opacity 0.25s;"
            >
              <svg
                height="100%"
                style="position: absolute; top: 0px;"
                viewBox="-2 -5 17 21"
                width="100%"
              >
                <path
                  d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
                  fill="#fff"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              style="height: 28px; width: 30px; position: absolute; opacity: 1; right: 0px; top: 0px; pointer-events: none; transition: opacity 0.25s;"
            >
              <svg
                height="100%"
                style="position: absolute; top: 0px;"
                viewBox="-2 -5 14 20"
                width="100%"
              >
                <path
                  d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
                  fill="#fff"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
          <div
            class="react-switch-handle"
            style="height: 26px; width: 26px; background: rgb(255, 255, 255); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(1px); top: 1px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
          />
          <input
            aria-checked="false"
            role="switch"
            style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app__header header col-12"
    >
      <span
        class="header__button"
      >
        LM
      </span>
      <ul
        class="header__nav"
      >
        <li
          class="header__nav-category"
        >
          Bio
        </li>
        <li
          class="header__nav-category"
        >
          Experience
        </li>
        <li
          class="header__nav-category"
        >
          Projects
        </li>
        <li
          class="header__nav-category"
        >
          Contact
        </li>
      </ul>
      <div
        class="header__switch"
        style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 14px; transition: opacity 0.25s; user-select: none;"
      >
        <div
          class="react-switch-bg"
          style="height: 28px; width: 56px; margin: 0px; position: relative; background: rgb(255, 137, 6); border-radius: 14px; cursor: pointer; transition: background 0.25s;"
        >
          <div
            style="height: 28px; width: 30px; position: relative; opacity: 0; pointer-events: none; transition: opacity 0.25s;"
          >
            <svg
              height="100%"
              style="position: absolute; top: 0px;"
              viewBox="-2 -5 17 21"
              width="100%"
            >
              <path
                d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
                fill="#fff"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <div
            style="height: 28px; width: 30px; position: absolute; opacity: 1; right: 0px; top: 0px; pointer-events: none; transition: opacity 0.25s;"
          >
            <svg
              height="100%"
              style="position: absolute; top: 0px;"
              viewBox="-2 -5 14 20"
              width="100%"
            >
              <path
                d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
                fill="#fff"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
        <div
          class="react-switch-handle"
          style="height: 26px; width: 26px; background: rgb(255, 255, 255); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(1px); top: 1px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
        />
        <input
          aria-checked="false"
          role="switch"
          style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
          type="checkbox"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
